docs/frontend/html-coding-style.md

```diff
@@ -0,0 +1,329 @@
+
+Here are some guidelines and tips for creating good HTML code:
+
+- Use HTML5 semantic tags [Semantic Elements](https://developer.mozilla.org/en-US/docs/Glossary/Semantics#semantic_elements)
+- Avoid an excessive [DOM size](https://developer.chrome.com/docs/lighthouse/performance/dom-size/?utm_source=lighthouse&utm_medium=lr)
+- Follow accessibility best practices [HTML and accessibility](https://developer.mozilla.org/en-US/docs/Learn/Accessibility/HTML)
+- HTML5 code should be valid. Check your code on [W3C Validator](https://validator.w3.org/)
+
+## Formatting
+
+It is recommended to use Prettier as a code formatter to keep the code style consistent.
+
+[Here is an example](assets/.prettierrc.json) of our configuration file to learn about the current rules, and read the [Prettier documentation](https://prettier.io/docs/en/index.html).
+
+Prettier formats all the code and keeps the style consistent. Nevertheless, there are a few additional rules that you need to follow.
+
+Also, you can use [EditorConfig](https://editorconfig.org/) file for your Code Editor. Here is an example file of [.editorconfig](assets/.editorconfig).
+
+## Doctype
+
+You should use the [HTML5](https://dev.w3.org/html5/spec-LC/) doctype. It is short, easy to remember, and backwards compatible.
+
+## Document characterset
+
+You should also define your document's characterset like so:
+
+```HTML
+<meta charset="utf-8" />
+```
+
+## Document Language
+
+Set the document language using the [lang](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes#lang) attribute on your [<html>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/html) element:
+
+```HTML
+
+<html lang="en-US"></html>
+
+```
+
+## Viewport Meta Tag
+
+You should include the following [<meta>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta) viewport element in all your web pages:
+
+```HTML
+
+<meta name="viewport" content="width=device-width" />
+
+```
+
+## Indents
+
+- **Don't use tabs** to indent text; use spaces only. Different text editors interpret tabs differently, and some Markdown features expect spaces and not tabs.
+- **Indent by two spaces** per indentation level.
+- **Don't leave trailing spaces** at the end of a line (except as needed for Markdown).
+
+## Line Length
+
+Break lines at 80 characters except in the following cases:
+
+- The metadata tags at the top of files (such as page.metaDescription) have to be all on one line, so those lines can be as long as needed.
+- If a URL in a link has a line break, the link won't work. If a URL is longer than 80 characters (quite common), you're stuck with it. In that case, put the URL on its own line with the href attribute to make it easier to review the text before and after, as the following example shows:
+
+```HTML
+You can find more information in
+<a href="https://example.com/long-url/johan-gambolputty-de-von-ausfern-…-von-hautkopf-of-ulm.html"
+class="external">his biography.</a>
```

Comment:
would be great to write separate section on lines wrap - rules how to right multiline tags correctly, like
```
<div
  attr="1"
>
  inner hrml
</div>
```

(or mabe you have different formatting for multiline)